src/utils/git.cpp

```diff
@@ -79,7 +79,25 @@ void Utils::Git::pushCurrentNoteFolder() {
     auto* process = new QProcess();
     process->setWorkingDirectory(NoteFolder::currentLocalPath());
 
+    if (!executeGitCommand(git, QStringList{"pull", "--no-commit", "--no-ff"}, process)) {
+        /* pull with merge failed */
+        executeGitCommand(git, QStringList{"merge", "--abort"}, process);
+
+        Utils::Gui::warning(
+                nullptr, QObject::tr("Git pull/merge failed!"),
+                QObject::tr("The pull and merge of the remote repository failed. Please check on the command line whether the remote is configured correctly"
+                    "or if a manual merge is required."),"git-pull-merge-failed.");
+        delete(process);
+        return;
+    } else {
+        executeGitCommand(git, QStringList{"commit", "-am", "Merge commit"}, process, false, true);
+    }
+
     if (!executeGitCommand(git, QStringList{"push", "-q"}, process)) {
+        Utils::Gui::warning(
```

Comment:
Again, this will spam the user with error dialogs...

Reply:
Moving this to the status bar?

Reply:
It will be hard to do that from a library (`git.cpp`).

Reply:
I pass a pointer from the mainwindow in, when the commit or pull is called.